Remove unused functions from `LegacyInlineTextBox.cpp|h`

Ahmad-S792 opened the pull request:
#### ccb1f4262e9fe058cdf4cf146326df2642f3a529
<pre>
Remove unused functions from `LegacyInlineTextBox.cpp|h`
<a href="https://bugs.webkit.org/show_bug.cgi?id=287334">https://bugs.webkit.org/show_bug.cgi?id=287334</a>
<a href="https://rdar.apple.com/144450111">rdar://144450111</a>

Reviewed by Alan Baradlay.

This patch just removes unused functions and continue on path
of Legacy Layout clean-up.

* Source/WebCore/rendering/LegacyInlineTextBox.cpp:
(WebCore::LegacyInlineTextBox::logicalOverflowRect const): Deleted.
* Source/WebCore/rendering/LegacyInlineTextBox.h:
(WebCore::LegacyInlineTextBox::logicalTopVisualOverflow const): Deleted.
(WebCore::LegacyInlineTextBox::logicalBottomVisualOverflow const): Deleted.
(WebCore::LegacyInlineTextBox::logicalLeftVisualOverflow const): Deleted.
(WebCore::LegacyInlineTextBox::logicalRightVisualOverflow const): Deleted.

Canonical link: <a href="https://commits.webkit.org/290114@main">https://commits.webkit.org/290114@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b9cc657b4793bcce07647e38ad514c30604ba83e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/88965 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/8489 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/43686 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/93937 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/39725 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/91016 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/8876 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/16673 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/68539 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/26214 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/91967 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/6769 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/80649 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/48903 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/6518 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/35049 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/38833 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/76880 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/35999 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/95776 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/16145 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/11787 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/77418 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/16401 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/76457 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/76706 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/18915 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/21094 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/19676 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/9226 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/16159 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/21470 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/15900 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/19351 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/17681 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

alanbaradlay reviewed (approved):
We should put very little effort into cleaning up legacy at this point as Antti is going to make all of it go away in one go (unless there's some additional benefit(.

nt1m commented on Source/WebCore/rendering/LegacyInlineTextBox.cpp:
```diff
@@ -114,13 +114,6 @@ void LegacyInlineTextBox::markDirty(bool dirty)
-{
-    if (knownToHaveNoOverflow() || !gTextBoxesWithOverflow)
-        return enclosingIntRect(logicalFrameRect());
-    return gTextBoxesWithOverflow->get(this);
```
setLogicalOverflowRect, gTextBoxesWithOverflow can all be removed too it seems.

Ahmad-S792 replied:
'setLogicalOverflowRect' is used in LegacyInlineFlowBox.cpp, while 'gTextBoxesWithOverflow' is used in other areas. I would let Antti clean-up in one swoop.

nt1m replied:
> 'setLogicalOverflowRect' is used in LegacyInlineFlowBox.cpp

That call site does nothing, since nothing reads from `logicalOverflowRect()`. Ditto for `gTextBoxesWithOverflow`...

Commit:
Remove unused functions from `LegacyInlineTextBox.cpp|h`
https://bugs.webkit.org/show_bug.cgi?id=287334
rdar://144450111

Reviewed by Alan Baradlay.

This patch just removes unused functions and continue on path
of Legacy Layout clean-up.

* Source/WebCore/rendering/LegacyInlineTextBox.cpp:
(WebCore::LegacyInlineTextBox::logicalOverflowRect const): Deleted.
* Source/WebCore/rendering/LegacyInlineTextBox.h:
(WebCore::LegacyInlineTextBox::logicalTopVisualOverflow const): Deleted.
(WebCore::LegacyInlineTextBox::logicalBottomVisualOverflow const): Deleted.
(WebCore::LegacyInlineTextBox::logicalLeftVisualOverflow const): Deleted.
(WebCore::LegacyInlineTextBox::logicalRightVisualOverflow const): Deleted.

Canonical link: https://commits.webkit.org/290114@main
